src/ln/msgs.rs

```diff
@@ -1075,13 +1076,60 @@ impl MsgEncodable for UpdateFee {
 }
 
 impl MsgDecodable for ChannelReestablish {
-	fn decode(_v: &[u8]) -> Result<Self, DecodeError> {
-		unimplemented!();
+	fn decode(v: &[u8]) -> Result<Self, DecodeError> {
+		// TODO savil. Remove comment, after PR discussion.
+		// Why is PublicKey len == 33, shouldn't it be 64? https://git.io/fNWqR
```

Comment:
care to shed light on this question?

Reply:
No, all pubkeys are encoded in "compressed form", see https://docs.rs/secp256k1/0.9.2/secp256k1/key/struct.PublicKey.html#method.serialize